RMextract/getIONEX.py

```diff
@@ -680,51 +714,80 @@ def get_urllib_IONEXfile(time="2012/03/23/02:20:10.01",
+                try:
+                    secondary = request.urlopen(backupserver,timeout=30)
+                    backupfound = True
+                    server=backupserver
+                    logging.warning(f"Primary IONEX host '{server}' resolution "
+                                                f"failure. Trying backup at '{backupserver}'")
+                except Exception as e:
+                    logging.error(f"Primary and Backup Server not responding: {e}") #enable in lover environment
+    
+    
+    if isinstance(formatter, str):
+        try:
+            formatter = KNOWN_FORMATTERS[formatter]
+        except KeyError:
+            raise ValueError(f"Unknown formatter {formatter} - please provide a callable")
+    if not formatter:
+        # Check known servers
+        for known_server in KNOWN_FORMATTERS.keys():
+            if known_server in server:
+                formatter = KNOWN_FORMATTERS.get(known_server)
+                break
+        if not formatter:
+            raise ValueError(f"Unknown server {server} - please provide a formatter")
+    
+    url = formatter(server=server, prefix=prefix, year=year, dayofyear=dayofyear, yy=yy)
+
+    logging.info(f"Constructed {url=}.")
 
-    try:
-        primary = request.urlopen(server,timeout=30)
-        serverfound = True
-    except:
-            try:
-                secondary = request.urlopen(backupserver,timeout=30)
-                backupfound = True
-                server=backupserver
-                logging.warning(f"Primary IONEX host '{server}' resolution "
-                                            f"failure. Trying backup at '{backupserver}'")
-            except:
-                logging.error('Primary and Backup Server not responding') #enable in lover environment
-    if "http://ftp.aiub.unibe.ch" in server:
-        url = "http://ftp.aiub.unibe.ch/CODE/%4d/%s%03d0.%02dI.Z"%(year,prefix.upper(),dayofyear,yy)
-    elif "http://cddis.gsfc.nasa.gov" in server:
-        url = "http://cddis.gsfc.nasa.gov/gnss/products/ionex/%4d/%03d/%s%03d0.%02di.Z"%(year,dayofyear,prefix,dayofyear,yy)
-    elif "igsiono.uwm.edu.pl" in server:
-        url = "https://igsiono.uwm.edu.pl/data/ilt/%4d/igrg%03d0.%02di"%(year,dayofyear,yy)
 
     # Download IONEX file, make sure it is always uppercase
     fname = outpath+'/'+(url.split('/')[-1]).upper()
-    try:
-        site = request.urlopen(url,timeout=30)
-    except:
-        logging.info("No files found on %s for %s",server,fname)
-        return -1
-
-    output=open(fname,'wb')
-    output.write(site.read())
-    output.close()
-    ###### gunzip files
-    if fname[-2:].upper()==".Z":
-        command = "gunzip -dc %s > %s" % (fname, fname[:-2])
-        retcode = os.system(command)
-        if retcode:
-            raise RuntimeError("Could not run '%s'" % command)
-        else:
-            os.remove(fname)
-        fname=fname[:-2]
+    out_fname = fname[:-2] if fname[-2:].upper() == '.Z' else fname    
```

Comment:
I think the new IONEX filename format uses .gz instead of .Z, maybe we can already anticipate that here?